wicket-core/src/main/java/org/apache/wicket/markup/head/OnDomReadyHeaderItem.java

```diff
@@ -86,8 +86,8 @@ public void render(Response response)
 		CharSequence js = getJavaScript();
 		if (Strings.isEmpty(js) == false)
 		{
-			JavaScriptUtils.writeJavaScript(response, "Wicket.Event.add(window, \"domready\", " +
-				"function(event) { " + js + ";});");
+			JavaScriptUtils.writeJavaScript(response,
+				"var f = function() {" + js + ";};\nif ('loading' !== document.readyState) f(); else document.addEventListener('DOMContentLoaded', f);");
```

Comment:
I would like to change this code to be:
```
{ //scope
    const f = function() {" + js + ";}; ....
}
```

This way multiple `OnDomReadyHeader` items will work as expected in deferred mode
WDYT?

Reply:
That does no harm +1